src/test/java/io/vantiq/client/intg/VantiqIntegrationTest.java

```diff
@@ -334,6 +334,47 @@ public void testDeleteOne() throws Exception {
         assertThat("Delete removed record", handler.getBodyAsList().size(), is(0));
     }
 
+    @Test
+    public void testPublishServiceEvent() throws Exception {
+        Date now = new Date();
+
+        String id = "PB-" + now.getTime();
+        JsonObject message = new JsonObject();
+        message.addProperty("id", id);
+        message.addProperty("ts", getISOString(now));
+        message.addProperty("x", 3.14159);
+        message.addProperty("k", 42);
+
+        JsonObject embedded = new JsonObject();
+        embedded.addProperty("a", 1);
+        embedded.addProperty("b", 2);
+        message.add("o", embedded);
+
+        // Publish to service event.
```

Comment:
Here, we "cheat" a bit.  The following explains what's actually going on as doing things this way simplifies the test setup.  Don't need some app, etc., to respond.  We just use the plumbing.